pyccel/codegen/printing/fcode.py

```diff
@@ -1408,32 +1412,68 @@ def _print_FunctionDef(self, expr):
             sig = 'elemental {}'.format(sig)
 
         arg_code  = ', '.join(self._print(i) for i in chain( expr.arguments, out_args ))
+
+        imports = ''.join(self._print(i) for i in expr.imports)
+        prelude = ''.join(self._print(i) for i in args_decs.values())
+        parts = ["{}({}) {}\n".format(sig, arg_code, func_end,),
+                imports,
+                "implicit none\n",
+                prelude]
+        return '\n'.join(a for a in parts if a)
+
+    def _print_FunctionDef(self, expr):
+        self._handle_fortran_specific_a_prioris(list(expr.local_vars) +
+                                                list(expr.arguments)  +
+                                                list(expr.results))
+        # ... we don't print 'hidden' functions
+        if expr.hide:
+            return ''
+
+        name = self._print(expr.name)
+        self.set_current_function(name)
+
+        if expr.cls_name:
+            for k, m in list(_default_methods.items()):
+                name = name.replace(k, m)
+
+            cls_name = expr.cls_name
+            if not (cls_name == '__UNDEFINED__'):
+                name = '{0}_{1}'.format(cls_name, name)
+        else:
+            for i in _default_methods:
+                # because we may have a class Point with init: Point___init__
+                if i in name:
+                    name = name.replace(i, _default_methods[i])
+
+        signature = self.function_signature(expr, name)
+        decs = OrderedDict()
+        functions = expr.functions
+        func_interfaces = '\n'.join(self._print(i) for i in expr.interfaces)
```

Comment:
Has this been tested? I am not sure if we are allowed to write an interface in a function

Reply:
can you please tell me why we are not allowed to write an interface in a function. if this is the case could you also give an alternative solution.

Reply:
I don't know if we are allowed or not. That is why I am asking if it has been tested. If we are not allowed then it would be due to the fortran standard. It is sometimes quite picky about where objects can be placed (e.g. all variables must be declared at the start of a function, you can have a function in a function, but not  a function in a function in a function).
If you have tested it and it works then we should be ok.

Reply:
yes I think it's allowed

Reply:
Does that mean you've tested it?

Reply:
yes, all the tests uses a generated code that contains interfaces inside functions

Reply:
Ok, perfect